tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py

```diff
@@ -105,83 +105,119 @@ def test_label_swirl(self, xp, scp):
     'op': ['sum', 'mean', 'variance', 'standard_deviation'],
 }))
 @testing.with_requires('scipy')
-class TestNdimage(unittest.TestCase):
-
-    @testing.for_all_dtypes(no_bool=True, no_complex=True)
-    @testing.numpy_cupy_array_almost_equal(scipy_name='scp')
-    def test_ndimage_single_dim(self, xp, scp, dtype):
-        image = xp.arange(100, dtype=dtype)
-        label = testing.shaped_random((100,), xp, dtype=xp.int32, scale=3)
-        index = xp.array([0, 1, 2])
-        return getattr(scp.ndimage, self.op)(image, label, index)
-
-    @testing.for_all_dtypes(no_bool=True, no_complex=True)
-    @testing.numpy_cupy_array_almost_equal(scipy_name='scp')
-    def test_ndimage_multi_dim(self, xp, scp, dtype):
-        image = xp.arange(512, dtype=dtype).reshape(8, 8, 8)
-        label = testing.shaped_random((8, 8, 8), xp, dtype=xp.int32, scale=3)
-        index = xp.array([0, 1, 2])
-        return getattr(scp.ndimage, self.op)(image, label, index)
-
-    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+class TestStats(unittest.TestCase):
+
+    def _make_image(self, shape, xp, dtype):
+        if dtype == xp.bool_:
+            return testing.shaped_random(shape, xp, dtype=xp.bool_)
+        else:
+            return testing.shaped_arange(shape, xp, dtype=dtype)
+
+    @testing.for_all_dtypes(no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_single_dim(self, xp, scp, dtype):
+        image = self._make_image((100,), xp, dtype)
+        labels = testing.shaped_random((100,), xp, dtype=xp.int32, scale=4)
+        index = xp.array([1, 2, 3])
+        op = getattr(scp.ndimage, self.op)
+        return op(image, labels, index)
+
+    @testing.for_all_dtypes(no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_multi_dim(self, xp, scp, dtype):
+        image = self._make_image((8, 8, 8), xp, dtype)
+        labels = testing.shaped_random((8, 8, 8), xp, dtype=xp.int32, scale=4)
+        index = xp.array([1, 2, 3])
+        op = getattr(scp.ndimage, self.op)
+        return op(image, labels, index)
+
+    @testing.for_all_dtypes(no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_zero_dim(self, xp, scp, dtype):
+        image = self._make_image((), xp, dtype)
+        labels = testing.shaped_random((), xp, dtype=xp.int32, scale=4)
+        index = xp.array([1, 2, 3])
+        op = getattr(scp.ndimage, self.op)
+        return op(image, labels, index)
+
+    @testing.for_all_dtypes(no_complex=True)
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_ndimage_zero_dim(self, xp, scp, dtype):
+        image = self._make_image((), xp, dtype)
+        label = testing.shaped_random((), xp, dtype=xp.int32, scale=4)
+        index = xp.array([1, 2, 3])
+        op = getattr(scp.ndimage, self.op)
+        return op(image, label, index)
```

Comment:
It seems this function is a duplicate of the above one?

Reply:
Let me remember what I wanted to do...

Reply:
I removed the duplicate test.